tick/solver/tests/serializing_test.py

```diff
@@ -0,0 +1,81 @@
+
+# License: BSD 3 clause
+
+
+import io, unittest
+import numpy as np
+
+import dill as pickle
+
+from tick.solver.tests import TestSolver
+
+from tick.prox import ProxL1
+from tick.linear_model import ModelLinReg, SimuLinReg
+from tick.linear_model import ModelLogReg, SimuLogReg
+from tick.linear_model import ModelPoisReg, SimuPoisReg
+from tick.linear_model import ModelHinge, ModelQuadraticHinge, ModelSmoothedHinge
+
+from tick.robust import ModelAbsoluteRegression, ModelEpsilonInsensitive, ModelHuber, \
+                        ModelLinRegWithIntercepts, ModelModifiedHuber
+
+from tick.solver import AdaGrad, SGD, SDCA, SAGA, SVRG
+
+from tick.simulation import weights_sparse_gauss
+
+class Test(TestSolver):
+    def test_solver_saga(self):
+        ratio = 0.5
+        l_enet = 1e-2
+        sd = ratio * l_enet
+
+        solvers = [
+          AdaGrad(step=1e-3, max_iter=100, verbose=False, tol=0),
+          SGD    (step=1e-3, max_iter=100, verbose=False, tol=0),
+          SDCA   (l_l2sq=sd, max_iter=100, verbose=False, tol=0),
+          SAGA   (step=1e-3, max_iter=100, verbose=False, tol=0),
+          SVRG   (step=1e-3, max_iter=100, verbose=False, tol=0)
+        ]
+        model_map = {
+            ModelLinReg : SimuLinReg,
+            ModelLogReg : SimuLogReg,
+            ModelPoisReg: SimuPoisReg,
+            ModelHinge  : SimuLogReg,
+            ModelQuadraticHinge : SimuLogReg,
+            ModelSmoothedHinge : SimuLogReg,
+
+            ModelAbsoluteRegression : SimuLinReg,
+            ModelEpsilonInsensitive : SimuLinReg,
+            ModelHuber              : SimuLinReg,
+            # ModelLinRegWithIntercepts : SimuLinReg, # has coeffe issue?
```

Comment:
See https://x-datainitiative.github.io/tick/modules/robust.html#example to have an example of data simulated for `ModelLinRegWithIntercepts`

Reply:
k